dataprofiler/tests/profilers/test_int_column_profile.py

```diff
@@ -788,3 +788,35 @@ def test_profile_merge_bin_edges_indices(self):
         profile_2.update(data_2)
 
         profile_1 + profile_2
+
+    def test_diff(self):
+        """
+        Makes sure the IntColumn Diff() works appropriately.
+        """
+        data = [2, 'not an int', 6, 4]
+        df = pd.Series(data).apply(str)
+        profiler1 = IntColumn("Int")
+        profiler1.update(df)
+
+        data = [1, 15]
+        df = pd.Series(data).apply(str)
+        profiler2 = IntColumn("Int")
+        profiler2.update(df)
+
+        # Assert the difference report is correct
+        expected_diff = {
+            'max': -9.0,
+            'mean': -4.0,
+            'min': 1.0,
+            'stddev': -7.899494936611665,
+            'sum': -4.0,
+            'variance': -94.0
+        }
+        self.assertDictEqual(expected_diff, profiler1.diff(profiler2))
+        
+        # Assert type error is properly called
+        with self.assertRaises(TypeError) as exc:
+            profiler1.diff("Inproper input")
+        self.assertEqual(str(exc.exception),
+                         "Unsupported operand type(s) for diff: 'IntColumn' and"
+                         " 'str'")
```

Comment:
Should you also add this test in `test_numeric_stats_mixin_profile`?

Reply:
Ultimately, I agree this is an issue bc we have to repeat the tests for anything inheriting. However, I think this needs to be addressed in a test refactor for the mixin all at once. Additionally, if we ultimately merge the int / float profiles similar to how text can be string / text, this would also have an impact on this change.